internals/search_queries.py

```diff
@@ -394,3 +408,45 @@ def sorted_by_review_date(descending: bool) -> Future:
     'gate.requested_on': sorted_by_pending_request_date,
     'gate.reviewed_on': sorted_by_review_date,
     })
+
+
+def query_any_start_milestone(
+  operator: str,
+  val_list: list[QueryValue | Interval[QueryValue]],
+  limit: int | None = None,
+) -> Future:
+  """Finds features shipping at any stage to any platform in milestones matching the val_list."""
+  disjunction = []
+  for field_name in [
+    'browsers.chrome.android',
+    'browsers.chrome.desktop',
+    'browsers.chrome.devtrial.android.start',
+    'browsers.chrome.devtrial.desktop.start',
+    'browsers.chrome.devtrial.ios.start',
+    'browsers.chrome.devtrial.webview.start',
+    'browsers.chrome.ios',
+    'browsers.chrome.ot.android.start',
+    'browsers.chrome.ot.desktop.start',
+    'browsers.chrome.ot.ios.start',
+    'browsers.chrome.ot.webview.start',
+    'browsers.chrome.webview',
+  ]:
+    stage_types = [
+      st for st in STAGE_TYPES_BY_QUERY_FIELD[field_name].values() if st is not None
+    ]
+    disjunction.append(
+      ndb.AND(
+        Stage.stage_type.IN(stage_types),
+        build_filter(STAGE_QUERIABLE_FIELDS[field_name], operator, val_list),
+      )
+    )
+  return Stage.query(ndb.OR(*disjunction)).fetch_async(
```

Comment:
Datastore (and Firestore in Datastore mode) requires an index that exactly matches the set of conditions in the query.  So, you'll need to add something to `index.yaml`.  The local datastore emulator does not require this.

It can be tricky to create the right index, so the way that I often do this is to run in staging and look at the error message that is generated.

Reply:
The new query is covered by the 
```yaml
- kind: Stage
  properties:
  - name: stage_type
  - name: milestones.desktop_first
  - name: feature_id
```
indices that support the existing filters that make up this disjunction. I checked by running the datastore emulator in persistent mode and looking at the index.yaml file that it generated.